File: examples/csharp/print_render_extension.cs
using System;
using libsbgncs;

public class print_render_extension
{

static void printRenderExtension(SbgnMap map)
{
  if (map == null)
  {
    Console.WriteLine(" Map was null ..." );
    return;
  }

  var info = map.getRenderInformation();
  if (info == null)
  {
    Console.WriteLine(" RenderInformation was null" );
    return;
  }

  Console.WriteLine("RenderInformation" );
  if (info.isSetId())
    Console.WriteLine("id: {0}", info.getId() );

  if (info.isSetName())
    Console.WriteLine("name: {0}", info.getName() );

  if (info.isSetProgramName())
    Console.WriteLine("program name: {0}", info.getProgramName() );

  if (info.isSetProgramVersion())
    Console.WriteLine("program version: {0}", info.getProgramVersion() );

  if (info.getNumColorDefinitions() > 0)
    Console.WriteLine("# color definitions: {0}", info.getNumColorDefinitions() );

  if (info.getNumGradientDefinitions() > 0)
    Console.WriteLine("# gradient definitions: {0}", info.getNumGradientDefinitions() );

  if (info.getNumLocalStyles() > 0)
    Console.WriteLine("# style definitions: {0}", info.getNumLocalStyles() );

}

public static int Main(string[] args)
{

  if (args.Length != 1)
  {
    Console.WriteLine(" usage: print_render_extension <filename>" );
    return 1;
  }

  var doc = libsbgn.readSBGNFromFile(args[0]);

  printRenderExtension(doc.getMap());


  return 0;
}

}

File: examples/c++/print_render_extension.cpp
#include <iostream>

#include <sbgn/SbgnTypes.h>
#include <sbml/packages/render/common/RenderExtensionTypes.h>

void printRenderExtension(SbgnMap* map)
{
  if (map == NULL)
  {
    std::cout << " Map was NULL ..." << std::endl;
    return;
  }

  auto* info = map->getRenderInformation();
  if (info == NULL)
  {
    std::cout << " RenderInformation was NULL" << std::endl;
    return;
  }

  std::cout << "RenderInformation" << std::endl;
  if (info->isSetId())
    std::cout << "id: " << info->getId() << std::endl;

  if (info->isSetName())
    std::cout << "name: " << info->getName() << std::endl;

  if (info->isSetProgramName())
    std::cout << "program name: " << info->getProgramName() << std::endl;

  if (info->isSetProgramVersion())
    std::cout << "program version: " << info->getProgramVersion() << std::endl;

  if (info->getNumColorDefinitions() > 0)
    std::cout << "# color definitions: " << info->getNumColorDefinitions() << std::endl;

  if (info->getNumGradientDefinitions() > 0)
    std::cout << "# gradient definitions: " << info->getNumGradientDefinitions() << std::endl;

  if (info->getNumLocalStyles() > 0)
    std::cout << "# style definitions: " << info->getNumLocalStyles() << std::endl;

}

int main(int argc, const char* argv[])
{

  if (argc != 2)
  {
    std::cout << " usage: print_render_extension <filename>" << std::endl;
    return 1;
  }

  auto* doc = readSBGNFromFile(argv[1]);

  printRenderExtension(doc->getMap(0));

  delete doc;

  return 0;
}